### ISSUE TITLE
Settings sync is suspended due to auth issues, yet signing in again doesn't work

### ISSUE THREAD

**Issue by josh-berry** [microsoft/vscode#130314]
Issue Type: <b>Bug</b>

After updating to the latest version of VSCode, I am now seeing the following error repeatedly on restart:

> Settings sync is suspended because of successive authorization failures. Please sign in again to continue synchronizing

Clicking the "Sign In" button shows the list of accounts, and shows my Microsoft account (which I am using for settings sync) as "Signed in". If I then click my Microsoft account, nothing happens, and the same error is shown again.

I was able to successfully sign out and sign in again, and that seems to have resolved the immediate problem, but something clearly wasn't in sync--some part of VSCode thought it couldn't sync, and some other part thought it was signed in successfully.

Mostly filing this as an FYI in case there's something here that needs to be looked into further.

VS Code version: Code 1.59.0 (379476f0e13988d90fab105c5c19e7abc8b1dea8, 2021-08-04T23:13:12.822Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No
Remote OS version: Linux x64 5.10.16.3-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 3900X 12-Core Processor             (24 x 3800)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|63.94GB (36.68GB free)|
|Process Argv|--crash-reporter-id 12a7846e-0844-43a9-a018-5c9919bb6fbd|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Alpine|
|OS|Linux x64 5.10.16.3-microsoft-standard-WSL2|
|CPUs|AMD Ryzen 9 3900X 12-Core Processor (24 x 3800)|
|Memory (System)|50.16GB (37.80GB free)|
|VM|0%|
</details><details><summary>Extensions (22)</summary>

Extension|Author (truncated)|Version
---|---|---
remote-containers|ms-|0.191.0
remote-ssh|ms-|0.65.7
remote-ssh-edit|ms-|0.65.7
remote-wsl|ms-|0.58.2
vscode-remote-extensionpack|ms-|0.21.0
svg-preview|Sim|2.8.3
rewrap|stk|1.14.0
pestfile|xor|0.1.0
npm-intellisense|chr|1.4.0
vscode-eslint|dba|2.1.23
gitlens|eam|11.6.0
vscode-npm-script|eg2|0.3.22
json-tools|eri|1.0.2
json-pretty-printer|eus|1.1.0
vscode-pull-request-github|Git|0.29.0
rust-doc-viewer|JSc|2.0.1
rust-analyzer|mat|0.2.694
git-graph|mhu|1.30.0
vetur|oct|0.34.1
vscode-coverage-gutters|rya|2.8.0
trailing-spaces|sha|0.3.1
code-spell-checker|str|1.10.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551:30345470
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805:30301674
pythonvspyt200:30340761
vscextlangct:30333562
binariesv615:30325510
vsccppwtct:30329789
pythonvssor306:30344512
bridge0708:30335490
vstre464:30346472

```

</details>

**Comment by Ajay-056** [comment]
I'm also getting this same error after updating to this version.

**Comment by SharakPL** [comment]
Got the same error, even though I was already signed in. The main problem here is it's not clear what to do with this error and all you need to do is to choose an acoount for Settings Sync (last option, opens browser to sign in).

**Comment by SharakPL** [comment]
@chrisant996 do you allow cookies in your browser? If not then it might be the reason you have to sign in every time.

**Comment by chrisant996** [comment]
> @chrisant996 do you allow cookies in your browser? If not then it might be the reason you have to sign in every time.

I allow cookies. It was working fine for months. New update, and suddenly I have to sign in multiple times in a _single vscode session_ because it keeps running into "successive sync failures". This is a vscode issue, not pebkac. 😉

**Comment by wasker** [comment]
I've seen this on multiple machines too.

Not sure how cookies in the browser relevant to this issue: VSCode uses browser to go through the OAuth process (to get some sort of token) after which browser is irrelevant.

**Comment by TylerLeonhardt** [comment]
The timeline of this lines up with an issue we were experiencing around this time. Can anyone confirm that they still get this:

> Settings sync is suspended because of successive authorization failures. Please sign in again to continue synchronizing

and if so, can you provide the contents of the Microsoft Authentication output pane:
![image](https://user-images.githubusercontent.com/2644648/128087382-c706864f-7005-4a2e-878b-662c69c4457e.png)

And also anything that shows up in Dev Tools:

* ctrl + shift + P
* "Dev Tools"

**Comment by SharakPL** [comment]
Haven't noticed any problems of this matter last couple weeks.

**Comment by dltmurphy** [comment]
I'm no longer getting the issue however i will check that output if it reoccurs

**Comment by TylerLeonhardt** [comment]
Since two of you aren't, I'm going to close this but if anyone _does_ see this issue still please don't hesitate to respond here or open a new issue.

**Comment by laurin1** [comment]
I had the issue yesterday and the only way to get rid of it was to sign into Github to use Live Share and that made it start working for Settings Sync as well.

**Comment by TylerLeonhardt** [comment]
@laurin1 when you see this again, please include the info I mentioned here: https://github.com/microsoft/vscode/issues/130314#issuecomment-933765583

**Comment by josh-berry** [comment]
Hit this issue again on the latest release:

> Version: 1.61.2 (user setup)
> Commit: 6cba118ac49a1b88332f312a8f67186f7f3c1643
> Date: 2021-10-19T14:57:20.575Z
> Electron: 13.5.1
> Chrome: 91.0.4472.164
> Node.js: 14.16.0
> V8: 9.1.269.39-electron.0
> OS: Windows_NT x64 10.0.22000

Recent output from the "Microsoft Authentication" output tab seems to show everything is fine:

```
[Info  - 20:57:46.804] Refreshing token...
[Info  - 20:57:47.208] Token refresh success
[Info  - 20:57:47.209] Token available from cache, expires in 4185999 milliseconds
[Info  - 22:07:03.217] Refreshing token...
[Info  - 22:07:03.789] Token refresh success
[Info  - 22:07:03.790] Token available from cache, expires in 5301999 milliseconds
[Info  - 22:57:03.598] Token available from cache, expires in 2302191 milliseconds
[Info  - 22:57:05.481] Token available from cache, expires in 2300308 milliseconds
[Info  - 23:16:43.383] Token available from cache, expires in 1122406 milliseconds
[Info  - 23:16:43.427] Token available from cache, expires in 1122362 milliseconds
[Info  - 23:17:18.689] Token available from cache, expires in 1087100 milliseconds
[Info  - 23:34:55.796] Refreshing token...
[Info  - 23:34:56.408] Token refresh success
[Info  - 23:34:56.409] Token available from cache, expires in 4938999 milliseconds
[Info  - 00:45:26.727] Token available from cache, expires in 708681 milliseconds
[Info  - 00:45:28.961] Token available from cache, expires in 706447 milliseconds
[Info  - 00:45:29.56] Token available from cache, expires in 706352 milliseconds
[Info  - 00:45:29.103] Token available from cache, expires in 706305 milliseconds
[Info  - 00:45:29.157] Token available from cache, expires in 706252 milliseconds
[Info  - 00:45:29.202] Token available from cache, expires in 706206 milliseconds
[Info  - 00:45:31.202] Token available from cache, expires in 704206 milliseconds
[Info  - 00:45:32.726] Token available from cache, expires in 702682 milliseconds
[Info  - 00:45:32.773] Token available from cache, expires in 702635 milliseconds
[Info  - 00:46:35.609] Token available from cache, expires in 639799 milliseconds
[Info  - 00:46:38.11] Token available from cache, expires in 637397 milliseconds
[Info  - 00:46:38.105] Token available from cache, expires in 637303 milliseconds
```

Developer tools console only shows the same message that appears in the notification, plus some error logs from an ESLint extension I have installed (number of occurrences of each message is in `[brackets]` at the beginning of each message):

```
[18] log.ts:301   ERR Request textDocument/codeAction failed with message: No ESLint configuration found in /home/des/src/tab-stash/src/model.: Error: Request textDocument/codeAction failed with message: No ESLint configuration found in /home/des/src/tab-stash/src/model.
	at /home/des/.vscode-server/extensions/dbaeumer.vscode-eslint-2.2.2/client/out/extension.js:1:52755
	at /home/des/.vscode-server/extensions/dbaeumer.vscode-eslint-2.2.2/client/out/extension.js:1:53049
	at Immediate.<anonymous> (/home/des/.vscode-server/extensions/dbaeumer.vscode-eslint-2.2.2/client/out/extension.js:1:53414)
	at processImmediate (internal/timers.js:461:21)

[5] notificationsAlerts.ts:42 Settings sync is suspended because of successive authorization failures. Please sign in again to continue synchronizing
onDidChangeNotification @ notificationsAlerts.ts:42
```
